### PR TITLE
Make Site as optional

### PR DESCRIPTION
**Pull request by Surgo**
In Django 1.11 and when not use `Site` framework, following error occured.

```
RuntimeError: Model class django.contrib.sites.models.Site doesn't declare an explicit app_label and isn't in an application in INSTALLED_APPS.
```

Use a shortcut.

### CODE REVIEW

**Comment by joshblum** [comment]
@Surgo thanks!! Will merge once tests pass :)

File: registration/models.py [hunk 1]
```diff
@@ -422,7 +422,7 @@ def _activate(self, profile, get_profile):
         """
 
         if not profile.user.is_active and not profile.activated:
-            site = Site.objects.get_current()
+            site = get_current_site()
```
Comment (joshblum) [inline registration/models.py]: you have to pass the `request` in https://docs.djangoproject.com/en/1.8/ref/contrib/sites/#django.contrib.sites.shortcuts.get_current_site :)

Comment (Surgo) [inline registration/models.py]: Oooooops. I'll fix it :(

Comment (Surgo) [inline registration/models.py]: OMG... May be it's heavy...

Comment (joshblum) [inline registration/models.py]: Does the fix not work?

Comment (Surgo) [inline registration/models.py]: If `SupervisedRegistrationManager._activate` interface is changed (like `_activate(self, site, profile, get_profile)`), have to change `RegistrationManager._activate` and `RegistrationManager.activate_user` as same as `SupervisedRegistrationManager`.

Its not have `requests` or `site`.

Comment (joshblum) [inline registration/models.py]: can you implement this fix?

Comment (Surgo) [inline registration/models.py]: Can I try to change `RegistrationManager.activate_user` interface to `activate_user(self, site, activation_key, get_profile=False)` like `RegistrationManager.create_inactive_user(self, site, new_user=None, send_email=True, request=None, profile_info={}, **user_info)`

Comment (Surgo) [inline registration/models.py]: I'll try tomorrow :)

Comment (joshblum) [inline registration/models.py]: I think that https://github.com/macropin/django-registration/blob/master/registration/admin.py#L37 can also be changed to use the `get_current_site` shortcut

Comment (Surgo) [inline registration/models.py]: Thank you. I will fix it too:)

**Comment by Surgo** [comment]
@joshblum Add `site` to `activate_user`'s arg :) and remove `Site`.

**Comment by joshblum** [comment]
@Surgo huge thanks for putting this together so quickly!!

**Pull request merged**